astropy/coordinates/jparser.py

```diff
@@ -0,0 +1,81 @@
+import re
+
+import numpy as np
+
+import astropy.units as u
+from astropy.coordinates import SkyCoord
+
+
+def _sexagesimal(g):
+    # convert matched regex groups to sexigesimal array
+    sign, h, m, s, frac = g
+    sign = [1, -1][sign == '-']
+    s = '.'.join((s, frac))
+    return sign * np.array([h, m, s], float)
+
+
+class Jparser(object):
+    """
+    Class that extracts coordinates from object names containing coordinated
+    eg: '2MASS J06495091-0737408'
+    """
+    basePattern = '([+-]?)(\d{1,2})(\d{2})(\d{2})\.?(\d{0,3})'
+    single_parser = re.compile(basePattern)
+    fullPattern = '(.*?J)' + basePattern * 2
+    parser = re.compile(fullPattern)
+
+    def __call__(self, name):
+        """Convert to `name` to `SkyCoords` object"""
+        return self.to_skycoord(name)
+
+    def match(self, name, raise_=False):
```

Comment:
Calling this `match` is confusing since it uses regex `search` underneath.  `match` implies that it starts at the beginning of the string.  It's also not the best idea to re-use your function/method name as a variable within that method.